src/lib/bootloader/udev_mapping.rb

```diff
@@ -34,17 +34,28 @@ def to_hash
     end
 
     # Converts full udev name to kernel device ( disk or partition )
-    # @param dev [String] device udev or kernel one like /dev/disks/by-id/blabla
+    # @param dev [String] device udev or kernel one like /dev/disk/by-id/blabla
     # @raise when device have udev format but do not exists
     def to_kernel_device(dev)
-      return dev if dev !~ /^\/dev\/disk\/by-/
-
-      all_devices[dev] or raise "Unknown udev device #{dev}"
+      # for non-udev devices try to see specific raid names (bnc#944041)
+      if dev =~ /^\/dev\/disk\/by-/
+        all_devices[dev] or raise "Unknown udev device #{dev}"
+      else
```

Comment:
The special case for devices starting with /dev/disk/by-... should not be needed since GetContVolInfo should also find those. But surely that is not tested.